addons/survey/models/survey_survey.py

```diff
@@ -809,6 +817,29 @@ def _get_survey_questions(self, answer=None, page_id=None, question_id=None):
             questions = questions & answer.predefined_question_ids
         return questions, page_or_question_id
 
+    def get_correct_answers(self, question_ids=False):
+        """ Returns a dictionary linking the survey question ids with their correct answers.
+        Limited to some questions if the question_ids parameter is specified. """
+        self.ensure_one()
+        domain = [('survey_id', '=', self.id), ('is_scored_question', '=', True)]
+        if question_ids:
+            domain.append(('id', 'in', question_ids))
+        questions = self.env['survey.question'].sudo().search(domain)
```

Comment:
This is a security issue: a public method that accesses records in sudo -> crac ! you leak answers. You can use "self" that is the current survey (and is called as sudo anyway in the controller), and use "filtered_domain" instead of manually calling search, like
```suggestion
        questions = self.question_ids.filtered_domain(domain)
```

The main change is: this method becomes private, effectively runs on the current survey, benefits from the cache, and does not use sudo anymore. It is the role of the caller to call with sudo() if necessary, once rights are granted.